tests/fixtures/s3.py

```diff
@@ -38,39 +68,107 @@
                 existing_buckets.append(storage_object.bucket_id)
             existing_objects.append(storage_object)
 
-s3_fixture = s3_fixture_factory(
-    existing_buckets=existing_buckets,
-    existing_objects=existing_objects,
+existing_buckets_ = (
+    DEFAULT_EXISTING_BUCKETS if existing_buckets is None else existing_buckets
+)
+non_existing_buckets_ = DEFAULT_NON_EXISTING_BUCKETS
+existing_objects_ = (
+    DEFAULT_EXISTING_OBJECTS if existing_objects is None else existing_objects
 )
+non_existing_objects_ = DEFAULT_NON_EXISTING_OBJECTS
+
+
+@dataclass
+class S3Fixture:
+    """Info yielded by the `s3_fixture` function"""
+
+    config: S3Config
+    storage: S3ObjectStorage
+    existing_buckets: List[str]
+    non_existing_buckets: List[str]
+    existing_objects: List[FileObject]
+    non_existing_objects: List[FileObject]
+
+
+async def populate_storage(
+    storage: S3ObjectStorage,
+    bucket_fixtures: List[str],
+    object_fixtures: List[FileObject],
+):
+    """Populate Storage with object and bucket fixtures"""
+
+    for bucket_fixture in bucket_fixtures:
+        await storage.create_bucket(bucket_fixture)
+
+    for object_fixture in object_fixtures:
+        if not await storage.does_bucket_exist(object_fixture.bucket_id):
+            await storage.create_bucket(object_fixture.bucket_id)
+
+        presigned_url = await storage.get_object_upload_url(
+            bucket_id=object_fixture.bucket_id, object_id=object_fixture.object_id
+        )
+
+        upload_file(
+            presigned_url=presigned_url,
+            file_path=object_fixture.file_path,
+            file_md5=object_fixture.md5,
+        )
+
+
+@pytest_asyncio.fixture
+async def s3_fixture() -> AsyncGenerator[S3Fixture, None]:
+    """Pytest fixture for tests depending on the ObjectStorageS3 DAO."""
+    with LocalStackContainer(image="localstack/localstack:0.14.2").with_services(
+        "s3"
+    ) as localstack:
+        config = config_from_localstack_container(localstack)
+        storage = S3ObjectStorage(config=config)
+        await populate_storage(
+            storage=storage,
+            bucket_fixtures=existing_buckets_,
+            object_fixtures=existing_objects_,
+        )
+
+        assert not set(existing_buckets_) & set(  # nosec
+            non_existing_buckets_
+        ), "The existing and non existing bucket lists may not overlap"
+
+        yield S3Fixture(
+            config=config,
+            storage=storage,
+            existing_buckets=existing_buckets_,
+            non_existing_buckets=non_existing_buckets_,
+            existing_objects=existing_objects_,
+            non_existing_objects=non_existing_objects_,
+        )
```

Comment:
This is following the old logic where the fixtures set a fixed initial state. We know have an s3_fixture directly in hexkit that exposes methods to the test for setting state. This makes the tests more readable and adds flexibility to the state.

Reply:
This is just moved out of the chassis-lib and simplified a bit.
Main goal was to remove chassis-lib.
I'd take care of that during other refactoring work, as mentioned above